### DISCUSSION TITLE
Unable to install Wazuh Agent

### DISCUSSION THREAD

**Discussion by wampfort** [wazuh/wazuh#25233] [Q&A] [answered]
Hi, can anyone help me with my issue installing the Wazuh agent?

I tried to install the agent on my Windows server and Ubuntu server, but I keep receiving the error shown below. I'm not sure what the problem is.

Thank you.
Ubuntu:
(Reading database ... 90208 files and directories currently installed.)
Preparing to unpack .../wazuh-agent_4.8.1-1_amd64.deb ...
Unpacking wazuh-agent (4.8.1-1) ...
dpkg-deb (subprocess): decompressing archive './wazuh-agent_4.8.1-1_amd64.deb' (size=10270680) member 'data.tar': internal gzip read error: '<fd:4>: invalid block type'
dpkg-deb: error: <decompress> subprocess returned error exit status 2
dpkg: error processing archive ./wazuh-agent_4.8.1-1_amd64.deb (--install):
 cannot copy extracted data for './var/ossec/lib/libwazuhext.so' to '/var/ossec/lib/libwazuhext.so.dpkg-new': unexpected end of file or stream
Errors were encountered while processing:
 ./wazuh-agent_4.8.1-1_amd64.deb

Windows server 2022:
Invoke-WebRequest : The underlying connection was closed: Could not establish trust relationship for the SSL/TLS
secure channel.
At line:1 char:1
+ Invoke-WebRequest -Uri https://packages.wazuh.com/4.x/windows/wazuh-a ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-WebRequest], WebExc
   eption
    + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand

**Comment by Damian-Mangold** [comment]
Hi @wampfort, To better assist you, could you please provide more details about the type of installation you performed? Did you follow any specific guide? If so, could you share it with me?

In the meantime, I share with you the official installation guide for Ubuntu (APT) and windows agents:
 - [Ubuntu](https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-linux.html)
 - [Windows](https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-windows.html)

**Reply by wampfort** [reply]
Hi Damian, Thank you for your reply. 

I received those error after doing this:
1. Go to Deploy new agent > Choose DEB amd64| MSI 32/64bits
2. Assigned my wazuh IP on "Assign a server address"
3. copy and paste what is on "Run the following commands to download and install the agent:"

I followed what is on the documentation website.
I installed it on ubuntu 22.04

https://documentation.wazuh.com/current/installation-guide/wazuh-indexer/step-by-step.html
https://documentation.wazuh.com/current/installation-guide/wazuh-server/step-by-step.html
https://documentation.wazuh.com/current/installation-guide/wazuh-dashboard/step-by-step.html

Also tried to follow the installation of agent here.
https://documentation.wazuh.com/current/installation-guide/wazuh-agent/wazuh-agent-package-linux.html
But encounter this error:
E: Conflicting values set for option Signed-By regarding source https://packages.wazuh.com/4.x/apt/ stable: /usr/share/keyrings/wazuh.gpg !=
E: The list of sources could not be read.

Thank you

**Reply by Damian-Mangold** [reply]
Ok, thanks for the reply. Let me do some local testing and see if I can reproduce the error.

**Reply by wampfort** [reply]
okay, thank you Damian.

**Reply by Damian-Mangold** [reply, marked as answer]
Hi @wampfort  I followed the guide you shared and was able to install the agent on Ubuntu 22.04.
![image](https://github.com/user-attachments/assets/6430db26-d104-48e4-b65c-33559bf1ce08)

---

On the other hand I have been able to reproduce the error:
![image](https://github.com/user-attachments/assets/9d629fe7-138e-4618-9329-fab901568288)

by running these commands, which only apply to Debian 7, 8, and Ubuntu 14:
![image](https://github.com/user-attachments/assets/09fce38e-7608-4863-9b2b-3922da6438c1)

To fix the problem edit the `/etc/apt/sources.list.d/wazuh.list` file and make sure there is only the following information associated with Wazuh:

![image](https://github.com/user-attachments/assets/61da0504-e28b-4a2b-a235-c8194e56a525)

Then install the agent with the command `WAZUH_MANAGER=‘10.0.0.2’ apt-get install wazuh-agent`.

---

Please note that to install the agent on Ubuntu 22.04 you only need to execute the following commands (root user):
- `curl -s https://packages.wazuh.com/key/GPG-KEY-WAZUH | gpg --no-default-keyring --keyring gnupg-ring:/usr/share/keyrings/wazuh.gpg --import && chmod 644 /usr/share/keyrings/wazuh.gpg`
- `echo "deb [signed-by=/usr/share/keyrings/wazuh.gpg] https://packages.wazuh.com/4.x/apt/ stable main" | tee -a /etc/apt/sources.list.d/wazuh.list`
- `apt-get update`
- `WAZUH_MANAGER="10.0.0.2" apt-get install wazuh-agent`
- `systemctl daemon-reload`
- `systemctl enable wazuh-agent`
- `systemctl start wazuh-agent`

**Reply by wampfort** [reply]
Hi @Damian-Mangold, Thank you so much for your quick response. I'll give that a try in a bit. I was busy with other tasks last week.

**Reply by wampfort** [reply]
I am now able to install the agent. Thank you so much @Damian

**Comment by Damian-Mangold** [comment]
Regarding the Ubuntu error, I would recommend the following actions.

---

The error you're encountering when trying to install the `wazuh-agent_4.8.1-1_amd64.deb` package is due to an issue with the integrity of the `.deb` file you're trying to install. Specifically, the error messages:

- `internal gzip read error: 'fd:4: invalid block type'`
- `unexpected end of file or stream`

suggest that the `.deb` file might be corrupted or incomplete.

Here are some steps you can follow to resolve this issue:

### 1. **Verify the integrity of the `.deb` file**
   - **Check if the `.deb` file is complete or corrupted**. If you downloaded it manually, try downloading it again from a reliable source. Ensure that the download completes without interruptions.

### 2. **Delete the file and re-download it**

### 3. **Try the installation again**

### 4. **Check disk space**
   - Ensure you have enough disk space for the installation:
     ```bash
     df -h
     ```

**Comment by sebasfalcone** [comment]
Inactive thread, please re-open it if necessary
